internal/v1/handler_blueprints.go

```diff
@@ -81,30 +75,67 @@ func (bb *BlueprintBody) RedactPasswords() {
 	}
 }
 
-func (u *User) setPassword(password *string) {
-	if password != nil {
-		if u.Password == nil {
-			u.Password = new(string)
-		}
-		*u.Password = *password
+// Merges Password or SshKey from other User struct to this User struct if it is not set
+func (u *User) MergeExisting(other User) {
+	if u.Password == nil {
+		u.Password = other.Password
+	}
+	if u.SshKey == nil {
+		u.SshKey = other.SshKey
 	}
 }
 
-func (u *User) setSshKey(sshKey *string) {
-	if sshKey != nil {
-		if u.SshKey == nil {
-			u.SshKey = new(string)
-		}
-		*u.SshKey = *sshKey
+// User must have name and non-empty password or ssh key
+func (u *User) hasValidValues() error {
+	nameRe := regexp.MustCompile(`\S+`)
```

Comment:
Nitpick: If you compile regexp into package-level variable, it is only compiled once during application start. Here, it is compiled every time a user is validated, which is a tad slower.

Reply:
moved to the top of the file next oblueprintNameRegex